projects/storefrontlib/src/cms-structure/page/component/handlers/web-component.handler.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  ElementRef,
+  Inject,
+  Injectable,
+  Injector,
+  PLATFORM_ID,
+  Renderer2,
+  ViewContainerRef,
+} from '@angular/core';
+import { Observable } from 'rxjs';
+import { ComponentHandler } from './component-handler';
+import { DOCUMENT, isPlatformBrowser } from '@angular/common';
+import { CxApiService } from '../services/cx-api.service';
+import { CmsComponentMapping, Priority } from '@spartacus/core';
+import { CmsComponentData } from '../../model';
+
+/**
+ * Component handler responsible for launching cms web components.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class WebComponentHandler implements ComponentHandler {
+  constructor(
+    @Inject(DOCUMENT) protected document: any,
+    @Inject(PLATFORM_ID) protected platform: any
+  ) {}
+
+  private loadedWebComponents: { [path: string]: any } = {};
+
+  hasMatch(componentMapping: CmsComponentMapping): boolean {
+    return (
+      typeof componentMapping.component === 'string' &&
+      componentMapping.component.includes('#')
+    );
+  }
+
+  getPriority(): Priority {
+    return Priority.LOW; // low, as it's a default matcher
+  }
+
+  launcher(
+    componentMapping: CmsComponentMapping,
+    viewContainerRef: ViewContainerRef,
+    elementInjector?: Injector
+  ): Observable<[ElementRef]> {
+    return new Observable<[ElementRef]>((observer) => {
+      let webElement;
+      let active = true;
```

Comment:
P2: Why do we need this `active` variable?

Reply:
To do proper clean up in case unsubsdciption happen before launcher will initialize